undertow/src/main/java/org/wildfly/extension/undertow/HostAdd.java

```diff
@@ -69,6 +70,7 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
         final boolean isDefaultHost = defaultServerName.equals(serverName) && name.equals(defaultHostName);
         final int defaultResponseCode = HostDefinition.DEFAULT_RESPONSE_CODE.resolveModelAttribute(context, model).asInt();
         final boolean enableConsoleRedirect = !HostDefinition.DISABLE_CONSOLE_REDIRECT.resolveModelAttribute(context, model).asBoolean();
+        DefaultDeploymentMappingProvider.instance().addMapping(defaultWebModule, serverName, name);
```

Comment:
I've added some validation and now #addMapping throws exception if there is duplicate registration.